frontend/src/components/features/chat/event-message.tsx

```diff
@@ -48,19 +48,50 @@ export function EventMessage({
 
   const { data: config } = useConfig();
 
-  // Use our query hook to check if feedback exists and get rating/reason
+  const shouldShowLikertScale = (
+    eventToCheck: OpenHandsAction | OpenHandsObservation,
+  ) => {
+    if (config?.APP_MODE !== "saas" || !isLastMessage) return false;
+
+    return (
+      isFinishAction(eventToCheck) ||
+      isErrorObservation(eventToCheck) ||
+      (isAssistantMessage(eventToCheck) && eventToCheck.action === "message")
+    );
+  };
```

Comment:
@OpenHands It seems that shouldShowLikertScale is no longer necessary, since we always need to add likertscale component to different places, e.g., if statement check for error action type. Can we simplify this a bit based on this observation:
1. remove shouldShowLikertScale
2. directly put renderLikertScale into if statements that check for error observations, assistant message, and error observation
3. inside renderLikertScale, only render it if it is in saas mode and is last message